inyoka/portal/tasks.py

```diff
@@ -78,3 +79,25 @@ def query_counter_task(cache_key, sql):
     cursor = connection.cursor()
     cursor.execute(sql)
     cache.set(cache_key, cursor.fetchone()[0])
+
+
+@shared_task
+def clean_privmsg_folders():
+    """
+    Remove all messages in private message folders (except 'archive')
+    after end of cache duration according to settings
+    has been reached.
+    """
+    privmsgs_trash = PrivateMessageEntry.objects.filter(
```

Comment:
I's better to move logic out of `@shared_task`, to  for example two methods in the model manger. So the shared task would only call these two (plus the logger)

This will make it possible to add some basic tests (for example: are only old messages are deleted? are messages in archive are kept? ...)